### PR TITLE
Implement `stop` method that effectively stops any networking

### PR DESCRIPTION
**Pull request by tomalec**
implements #248

### CODE REVIEW

File: test/specs/dom/index.js [hunk 1]
```diff
@@ -55,8 +55,11 @@ if (typeof window !== 'undefined') {
                     afterEach(function() {
                         window.history.pushState.restore();
                         historySpy = null;
+                        // stop all networking and DOM activity of abandoned instance
                         palindrom.unobserve();
                         palindrom.unlisten();
+                        palindrom.stop();
```
Comment (warpech) [inline test/specs/dom/index.js]: Seems that the following calls should be removed:

```
palindrom.unobserve();
palindrom.unlisten();
```

File: test/specs/dom/ot.js [hunk 2]
```diff
@@ -135,8 +135,11 @@ if (typeof window !== 'undefined') {
             assert.equal(palindrom.obj.newChildren.Name$, null);
             assert.deepEqual(palindrom.obj.newChildren, {});
 
+            // stop all networking and DOM activity of abandoned instance
             palindrom.unobserve();
             palindrom.unlisten();
+            palindrom.stop();
```
Comment (warpech) [inline test/specs/dom/ot.js]: Seems that the following calls should be removed:

```
palindrom.unobserve();
palindrom.unlisten();
```

**Comment by tomalec** [comment]
All addressed, could you take another look?

**Comment by warpech** [comment]
The commit looks good but it seems that you have pushed it on a wrong branch?

**Comment by tomalec** [comment]
🤦‍♂ sorry, I hope it's better now.

**Comment by warpech** [comment]
LGTM, however, one of the tests that you changed apparently is flaky, because e3f72000000eafdb5be9d49f7f9fa75cf42610d9 failed to build on Travis with the following error in Firefox:

```
x HTTP Client > Heartbeat > When created with `pingIntervalS: 2` > should send empty patch (`[]`) after 2 seconds after establish
Spec "HTTP Client > Heartbeat > When created with `pingIntervalS: 2` > should send empty patch (`[]`) after 2 seconds after establish" failed, the error was { actual: '0',
  name: 'AssertionError',
  showDiff: true,
  expected: '1',
  message: 'expected [] to have a length of 1 but got 0',
  stack: 'AssertionError@webpack://Tests/./node_modules/assertion-error/index.js?:74:13\nmodule.exports/Assertion.prototype.assert@webpack://Tests/./node_modules/chai/lib/chai/assertion.js?:141:13\nassertLength@webpack://Tests/./node_modules/chai/lib/chai/core/assertions.js?:2126:10\ndoAsserterAsyncAndAddThen@webpack://Tests/./node_modules/chai-as-promised/lib/chai-as-promised.js?:289:22\nmodule.exports/</</<@webpack://Tests/./node_modules/chai-as-promised/lib/chai-as-promised.js?:272:28\noverwritingChainableMethodWrapper@webpack://Tests/./node_modules/chai/lib/chai/utils/overwriteChainableMethod.js?:60:34\nchainableMethodWrapper@webpack://Tests/./node_modules/chai/lib/chai/utils/addChainableMethod.js?:113:49\n@webpack://Tests/./test/specs/client/heartbeat.js?:118:140\n' } AssertionError@webpack://Tests/./node_modules/assertion-error/index.js?:74:13
module.exports/Assertion.prototype.assert@webpack://Tests/./node_modules/chai/lib/chai/assertion.js?:141:13
assertLength@webpack://Tests/./node_modules/chai/lib/chai/core/assertions.js?:2126:10
doAsserterAsyncAndAddThen@webpack://Tests/./node_modules/chai-as-promised/lib/chai-as-promised.js?:289:22
module.exports/</</<@webpack://Tests/./node_modules/chai-as-promised/lib/chai-as-promised.js?:272:28
overwritingChainableMethodWrapper@webpack://Tests/./node_modules/chai/lib/chai/utils/overwriteChainableMethod.js?:60:34
chainableMethodWrapper@webpack://Tests/./node_modules/chai/lib/chai/utils/addChainableMethod.js?:113:49
@webpack://Tests/./test/specs/client/heartbeat.js?:118:140
```

Do you think this can be made less flaky?

**Pull request merged**